### ISSUE TITLE
Authentication framework

### ISSUE THREAD

**Issue by watermint** [watermint/toolbox#623]
tldr; Refactoring current file based structure to database based structure.

# Goal of change

To achieve ease of use, that requires reverse reference from peer name & scope combination or traversability from toolbox. For example, the program can show list of existing connection to Dropbox, then select from existing connection.

# Security

The current file based structure does not allow reverse reference from peer name & scope combination to the actual authentication file. That is design by default. But if the attacker can access all files under `secret` folder and they know the credential, attacker can easily traverse all token files. ([File and Directory Discovery](https://attack.mitre.org/techniques/T1083/))

So, considering for the security, the toolbox should protect from remote attack by absolute path like `$HOME/.toolbox/secret/abcdef.db`.

**Comment by watermint** [comment]
Token types:
* OAuth
* Basic auth
* Generated token

**Comment by watermint** [comment]
Portability:
- [ ] Can export partial data of existing database, then import/refer that database for unattended execution.

**Comment by watermint** [comment]
Database design

| column | description |
|--------|-------------|
| peer     | peer name   |
| service | service name (e.g. dropbox, google) |
| auth_type | auth type (e.g. oauth, basic) |
| auth_scope | scope parameter. JSON or other equiv. serialized text. this value must be equivalent if multiple scope with different order means same scope |
| secret | encrypted secret data (e.g. serialized oauth token, basic auth password) |

**Comment by watermint** [comment]
- [ ] Better error message on token expiration
- [ ] Token annotation

**Comment by watermint** [comment]
- [ ] Need to update command document by auth method
